packages/block-library/src/cover/save.js

```diff
@@ -64,7 +64,7 @@ export default function save( { attributes } ) {
 		{
 			'has-background-dim': dimRatio !== 0,
 			'has-parallax': hasParallax,
-			'has-background-gradient': customGradient,
+			'has-background-gradient': gradient || customGradient,
 			[ gradientClass ]: ! url && gradientClass,
```

Comment:
There is at least one condition here `! url` (not sure if `gradientClass` can be falsey) in determining whether a gradient class is applied. Should that also be taken into consideration above when considering to apply `has-background-gradient`?

Reply:
Hi @aduth when we have a gradient background, and a URL background we need to make the gradient background part of another element (we use a span just for that) and we apply the class to the other element that's why we use "! url && gradientClass". The idea of "has-background-gradient" is to allow targeting cover blocks that use a gradient background no matter if they also use an image background or not. I guess it would be possible to target the gradients using their gradients classes but themes can defined their own so one would need to use something like class attribute contains but that sounds hacky, also one would need to target the application of the class to this element or the span used when we don't have an image background. This class seems to make things easier.
The changes to the save make the save match what happens in the editor where we add the class provided that a gradient exists (custom or predefined). gradientClass will always be defined provided gradient attribute exists so "gradient || customGradient" or using "gradientClass || customGradient" is exactly the same.

Reply:
So in the serialized output:

https://github.com/WordPress/gutenberg/blob/52417c6b13143bef1105952ce404ac6683a978ab/packages/e2e-tests/fixtures/blocks/core__cover__deprecated-5.serialized.html#L2

It's expected that the `has-background-gradient` class is on a different element (`div`) than the `has-midnight-gradient-background` element (`span`), when a `url` is assigned?

Reply:
Yes it is expected our selectors already rely on this e.g:"&.has-background-dim:not(.has-background-gradient)::before,".